### PR TITLE
Avoid duplicating files

### PR DESCRIPTION
**Pull request by jakirkham**
Fixes up conflicts from style changes in PR ( https://github.com/conda-forge/conda-smithy/pull/1143 ).

cc @isuruf

Checklist
* [x] Added a ``news`` entry

### CODE REVIEW

**Comment by isuruf** [comment]
Thanks @jakirkham

File: conda_smithy/configure_feedstock.py [hunk 1]
```diff
@@ -480,9 +482,7 @@ def migrate_combined_spec(combined_spec, forge_dir, config):
 
     """
     combined_spec = combined_spec.copy()
-    migrations_root = os.path.join(
-        forge_dir, ".ci_support", "migrations", "*.yaml"
-    )
+    migrations_root = os.path.join(forge_dir, "migrations", "*.yaml")
```
Comment (CJ-Wright) [inline conda_smithy/configure_feedstock.py]:
```suggestion
    migrations_root = os.path.join(forge_dir, ".ci_support", "migrations", "*.yaml")
```
This needs to stay I think

Comment (jakirkham) [reply conda_smithy/configure_feedstock.py]: I'll defer to you and @isuruf. I don't know enough about what is happening here.

File: conda_smithy/configure_feedstock.py [hunk 2]
```diff
@@ -1252,7 +1256,7 @@ def _load_forge_config(forge_dir, exclusive_config_file):
         "provider": {
             "linux": "azure",
             "osx": "azure",
-            "win": "azure",
+            "win": "appveyor",
```
Comment (CJ-Wright) [inline conda_smithy/configure_feedstock.py]:
```suggestion
            "win": "azure",
```

File: conda_smithy/configure_feedstock.py [hunk 3]
```diff
@@ -1206,15 +1212,13 @@ def render_README(jinja_env, forge_config, forge_dir):
     with write_file(target_fname) as fh:
         fh.write(template.render(**forge_config))
 
-    code_owners_file = os.path.join(forge_dir, ".github", "CODEOWNERS")
     if len(forge_config["maintainers"]) > 0:
+        code_owners_file = os.path.join(forge_dir, ".github", "CODEOWNERS")
```
Comment (CJ-Wright) [inline conda_smithy/configure_feedstock.py]: This needs to go back to the way it was I think

File: conda_smithy/configure_feedstock.py [hunk 4]
```diff
@@ -1351,17 +1356,17 @@ def _load_forge_config(forge_dir, exclusive_config_file):
     logger.debug(log)
     logger.debug("## END CONFIGURATION\n")
 
-    if config["provider"]["linux_aarch64"] in {"default", "native"}:
+    for platform in ["linux_aarch64", "linux_armv7l"]:
```
Comment (CJ-Wright) [inline conda_smithy/configure_feedstock.py]: I think this block needs to go back.

File: conda_smithy/configure_feedstock.py [hunk 5]
```diff
@@ -1468,11 +1473,22 @@ def get_cfp_file_path(resolve=None, error_on_warn=True):
 def clear_variants(forge_dir):
     "Remove all variant files placed in the .ci_support path"
     if os.path.isdir(os.path.join(forge_dir, ".ci_support")):
-        configs = glob.glob(os.path.join(forge_dir, ".ci_support", "*.yaml"))
+        configs = glob.glob(os.path.join(forge_dir, ".ci_support", "*"))
```
Comment (CJ-Wright) [inline conda_smithy/configure_feedstock.py]: This might not b e needed?

File: conda_smithy/configure_feedstock.py (hunk 1 above)
Comment (CJ-Wright) [reply conda_smithy/configure_feedstock.py]: I'm just putting the two PRs side by side to see which changes were made. The other PR was not all ahead of master so some changes are actually reversions I think.

File: conda_smithy/configure_feedstock.py [hunk 6]
```diff
@@ -1537,9 +1553,9 @@ def main(
     )
 
     copy_feedstock_content(config, forge_dir)
-    if os.path.exists(os.path.join(forge_dir, "build-locally.py")):
-        set_exe_file(os.path.join(forge_dir, "build-locally.py"))
+    set_exe_file(os.path.join(forge_dir, "build-locally.py"))
```
Comment (CJ-Wright) [inline conda_smithy/configure_feedstock.py]: This might not be needed?

File: tests/test_configure_feedstock.py [hunk 7]
```diff
@@ -119,14 +127,15 @@ def test_r_matrix_azure(r_recipe, jinja_env):
...
+    # no appveyor.yaml should have been written.  Nothing else, either, since we only ran
+    #     appveyor render.  No matrix dir should exist.
     matrix_dir = os.path.join(r_recipe.recipe, ".ci_support")
     assert os.path.isdir(matrix_dir)
     # single matrix entry - readme is generated later in main function
     assert len(os.listdir(matrix_dir)) == 4
 
 
 def test_py_matrix_appveyor(py_recipe, jinja_env):
-    py_recipe.config["provider"]["win"] = "appveyor"
```
Comment (CJ-Wright) [inline tests/test_configure_feedstock.py]: I think this needs to stay in.

File: tests/test_configure_feedstock.py [hunk 8]
```diff
@@ -465,9 +476,7 @@ def test_migrator_downgrade_recipe(
         len(
             os.listdir(
                 os.path.join(
-                    recipe_migration_cfep9_downgrade.recipe,
-                    ".ci_support",
-                    "migrations",
+                    recipe_migration_cfep9_downgrade.recipe, "migrations"
```
Comment (CJ-Wright) [inline tests/test_configure_feedstock.py]: This needs to go back

File: tests/test_configure_feedstock.py [hunk 9]
```diff
@@ -500,9 +509,7 @@ def test_migrator_compiler_version_recipe(
         len(
             os.listdir(
                 os.path.join(
-                    recipe_migration_win_compiled.recipe,
-                    ".ci_support",
-                    "migrations",
+                    recipe_migration_win_compiled.recipe, "migrations"
```
Comment (CJ-Wright) [inline tests/test_configure_feedstock.py]: This also needs to go back

File: conda_smithy/configure_feedstock.py (hunk 1 above)
Comment (jakirkham) [reply conda_smithy/configure_feedstock.py]: Yeah I'm just saying I don't feel confident making these changes as I'm unclear what is happening. If you do, please feel free to just push them to this PR.

**Comment by CJ-Wright** [comment]
I'll come back to this today

**Comment by CJ-Wright** [comment]
@isuruf can you take a look at this to make certain I stayed true to your intent? Barring any issues I think this is ready for review/merge. @jakirkham

**Comment by jakirkham** [comment]
Thanks @CJ-Wright! That was very helpful. 😄

FWIW I tried merging this a different way and came to the same result that you did.

@isuruf, can you please double check when you have a chance?

File: tests/test_configure_feedstock.py [hunk 10]
```diff
@@ -28,6 +28,8 @@ def test_noarch_skips_travis(noarch_recipe, jinja_env):
     )
     # this configuration should be skipped
     assert not noarch_recipe.config["travis"]["enabled"]
+    # no appveyor.yaml should have been written.  Nothing else, either, since we only ran
+    #     appveyor render.  No matrix dir should exist.
```
Comment (isuruf) [inline tests/test_configure_feedstock.py]: This comment and following comments should be removed

**Comment by jakirkham** [comment]
Thanks. Addressed. Can you please take another look?

**Comment by isuruf** [comment]
Thanks for fixing this

**Comment by jakirkham** [comment]
Thanks for doing this work in the first place 🙂

**Pull request merged**